packages/firestore/test/unit/local/schema_migration.test.ts

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { expect } from 'chai';
+import { IndexedDbPersistence } from '../../../src/local/indexeddb_persistence';
+import {
+  ALL_STORES,
+  createOrUpgradeDb,
+  DbTarget,
+  DbTargetGlobal,
+  DbTargetGlobalKey,
+  V1_STORES
+} from '../../../src/local/indexeddb_schema';
+import { Deferred } from '../../../src/util/promise';
+import { SimpleDb, wrapRequest } from '../../../src/local/simple_db';
+import { IndexedDbQueryCache } from '../../../src/local/indexeddb_query_cache';
+import { PersistencePromise } from '../../../src/local/persistence_promise';
+
+const INDEXEDDB_TEST_DATABASE = 'schemaTest';
+
+function withDb(
+  schemaVersion,
+  fn: (db: IDBDatabase) => Promise<void>
+): Promise<void> {
+  return new Promise<IDBDatabase>((resolve, reject) => {
+    const request = window.indexedDB.open(
```

Comment:
I can't remember why Sebastian said we couldn't use SimpleDb.openOrCreate, but you might want to revisit, especially if we need it to create the PersistenceTransaction that gets passed to createOrUpgradeDb().

Reply:
It doesn't expose the version or the list of available object stores.

Reply:
Ahh, thanks (sorry I was being lazy and not looking back at that PR).  I think it would be fine to add that functionality if it'd be useful to us.

Reply:
I did not add this, as this test is the only place where the functionality would be required, and this is testing a very db-specific mechanism. If we end up with more tests needing this, we can revisit. Instead, I allowed creating a `SimpleDbTransaction` from an existing `IDBTransaction`, which gets us the API we want to use.